lib/omnibus/packagers/deb.rb

```diff
@@ -676,33 +676,42 @@ def remove(packages)
       shellout!("apt-get remove -y --force-yes #{packages}")
     end
 
-    def compression_level(val = nil)
-      unless val.nil?
+    def compression_level(val = NULL)
+      if null?(val)
+        @compression_level || 6
+      else
         unless val >= 0 && val <= 9
           raise InvalidValue.new(:compression_level, 'be an Integer between 0 and 9 included')
         end
+
+        @compression_level = val
       end
-      @compression_level = val || 6
     end
     expose :compression_level
 
-    def compression_threads(val = nil)
-      unless val.nil?
+    def compression_threads(val = NULL)
+      if null?(val)
+        @compression_threads || 1
```

Comment:
If my above comment makes sense, otherwise you can dismiss them all
```suggestion
        @compression_threads = 1
```

Reply:
I *think* the current code is correct, but I so understand your question & doubt :grimacing: